app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,374 @@
+import * as Utils from "libs/utils";
 import { updateUser } from "admin/admin_rest_api";
-import type { APIUserType } from "admin/api_flow_types";
+import { handleGenericError } from "libs/error_handling";
+import type { APIUserType, ExperienceDomainListType } from "admin/api_flow_types";
+import SelectExperienceDomain from "components/select_experience_domain";
+
+const { Column } = Table;
+
+type TableEntry = {
+  domain: string,
+  value: number,
+  lowestValue: number,
+  highestValue: number,
+  isShared: boolean,
+  changed: boolean,
+};
 
 type Props = {
-  onChange: Function,
-  onCancel: Function,
+  onChange: (Array<APIUserType>) => void,
+  onCancel: () => void,
   visible: boolean,
-  selectedUserIds: Array<string>,
-  users: Array<APIUserType>,
+  selectedUsers: Array<APIUserType>,
 };
 
 type State = {
-  domain: ?string,
-  level: ?string,
+  tableEntries: Array<TableEntry>,
+  removedDomains: Array<string>,
+  showOnlySharedExperiences: boolean,
 };
 
 class ExperienceModalView extends React.PureComponent<Props, State> {
-  state = {
-    domain: null,
-    level: null,
-  };
-
-  increaseExperience = (): void => {
-    this.setExperience(null, true);
-  };
-
-  setExperience = (event: ?SyntheticInputEvent<>, shouldAddValue: boolean = false): void => {
-    const { domain, level } = this.state;
-    if (domain && level !== null) {
-      const newUserPromises = this.props.users.map(user => {
-        if (this.props.selectedUserIds.includes(user.id)) {
-          let newExperienceLevel = parseInt(level);
-
-          if (shouldAddValue && user.experiences[domain]) {
-            newExperienceLevel = user.experiences[domain] + parseInt(level);
-          }
-
-          const newUser = update(user, {
-            experiences: { [domain]: { $set: newExperienceLevel } },
-          });
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      tableEntries: this.loadTableEntries(props.selectedUsers),
+      removedDomains: [],
+      showOnlySharedExperiences: false,
+    };
+  }
 
-          return this.sendUserToServer(newUser, user);
-        }
-        return Promise.resolve(user);
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.visible && !this.props.visible) {
+      this.setState({
+        tableEntries: this.loadTableEntries(nextProps.selectedUsers),
+        removedDomains: [],
+        showOnlySharedExperiences: false,
       });
+    }
+  }
+
+  sortEntries(entries: Array<TableEntry>): Array<TableEntry> {
+    return entries.sort(
+      Utils.localeCompareBy(([]: Array<TableEntry>), entry => entry.domain.toLowerCase()),
+    );
+  }
 
-      this.closeModal(newUserPromises);
+  loadTableEntries = (users: Array<APIUserType>): Array<TableEntry> => {
+    if (users.length <= 1) {
+      return this.sortEntries(
+        _.map(users[0].experiences, (value, domain) => ({
+          domain,
+          value,
+          lowestValue: -1,
+          highestValue: -1,
```

Comment:
But -1 probably means that the interval is not rendered in the table, right? If you add the comments to value, lowestValue, highestValue which I mentioned somewhere else, this will probably clear it up for me :)